server_vllm.py

```diff
@@ -334,9 +334,27 @@ async def completion_stream_generator(
         tool_choice, functions
     ) -> AsyncGenerator[str, None]:
         generator = wrap_vllm_generator(tool_choice=tool_choice)
+        tool_call_count = 0
         async for response in generate_openai_format_from_stream_async(
             generator, prompt_template, tool_choice
         ):
+            # Convert tool_calls to function_call if request.functions is provided
+            if (
+                functions
+                and len(functions) > 0
+                and "tool_calls" in response["delta"]
+                and response["delta"]["tool_calls"]
+                and len(response["delta"]["tool_calls"]) > 0
+            ):
+                tool_name = response["delta"]["tool_calls"][0]["function"]["name"]
+                tool_args = response["delta"]["tool_calls"][0]["function"]["arguments"]
+                if tool_name and len(tool_name) > 0 and tool_args == "":
+                    tool_call_count += 1
+            # Return finish_reason after the first tool_call is streamed if functions is provided
```

Comment:
I think need to write a function to convert format of tool_calls to function_call ? You can implement inside: 
functionary/prompt_template/prompt_utils.py
```
{
    "content": None,
    "function_call": None,
    "role": None if not return_role else "assistant",
    "tool_calls": [
        {
            "index": 0,
            "id": ...
            "function": {
                "arguments": xxx,
                "name": xxx
            },
            "type": "function"
        }
    ],
},
"finish_reason": finish_reason,
"index": 0,
}

-->

{
    "content": None,
    "function_call": {
        "name": xxx,
        "arguments": xxx
    },
    "role": None if not return_role else "assistant",
},
"finish_reason": finish_reason,
"index": 0,
}

```

Reply:
Only 1 place calls this function though. Is it necessary to create a function?

Reply:
Only 1 place handles converting chunk and 1 place handles converting non-streaming message.

Reply:
Yes, you don't need to write a function, but I didn't see the conversion for streaming, right ? should have something like this: https://github.com/MeetKai/functionary/blob/fix-tools-functions/server_vllm.py#L421
here